src/backend/distributed/commands/seclabel.c

```diff
@@ -0,0 +1,165 @@
+/*-------------------------------------------------------------------------
+ *
+ * seclabel.c
+ *
+ *    This file contains the logic of SECURITY LABEL statement propagation.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+
+#include "commands/seclabel.h"
+#include "distributed/commands.h"
+#include "distributed/commands/utility_hook.h"
+#include "distributed/coordinator_protocol.h"
+#include "distributed/deparser.h"
+#include "distributed/log_utils.h"
+#include "distributed/metadata_sync.h"
+#include "distributed/metadata/distobject.h"
+
+
+PG_FUNCTION_INFO_V1(citus_test_register_label_provider);
+
+
+/*
+ * citus_test_register_label_provider registers a dummy label provider
+ * named 'citus_tests_label_provider'. This is aimed to be used for testing.
+ */
+Datum
+citus_test_register_label_provider(PG_FUNCTION_ARGS)
+{
+	register_label_provider("citus_tests_label_provider", citus_test_object_relabel);
+	PG_RETURN_VOID();
+}
+
+
+/*
+ * PreprocessSecLabelStmt is executed before the statement is applied to the local
+ * postgres instance.
+ *
+ * In this stage we can prepare the commands that need to be run on all workers to assign
+ * security labels on distributed objects, currently supporting just Role objects.
+ */
+List *
+PreprocessSecLabelStmt(Node *node, const char *queryString,
+					   ProcessUtilityContext processUtilityContext)
+{
+	if (!IsCoordinator() || !ShouldPropagate())
+	{
+		return NIL;
+	}
+
+	SecLabelStmt *secLabelStmt = castNode(SecLabelStmt, node);
+
+	List *objectAddresses = GetObjectAddressListFromParseTree(node, false, false);
```

Comment:
as we're in the preprocess:

```suggestion
	List *objectAddresses = GetObjectAddressListFromParseTree(node, false, true);
```